### PR TITLE
feat(OpenAPI): Define API for DFSP Mock Login page

### PR DESCRIPTION
**Pull request by sridharvoruganti**
covers #[1967](https://github.com/mojaloop/project/issues/1967)
API definition for:
-  `POST /login`
-  `POST /authorize`
-  `POST /sendOTP`
-  `POST /verifyOTPorSecret`
-  `GET /OTPorSecret`

### CODE REVIEW

**Review by kleyow — APPROVED** [review]

File: src/interface/ttk-thirdparty-api-template.yaml [hunk 1]
```diff
@@ -1053,3 +1288,72 @@ components:
...
+          $ref: '#/components/schemas/UserName'
+        password:
+          $ref: '#/components/schemas/Password'
+        consentRequestId:
+          $ref: '../../node_modules/@mojaloop/api-snippets/v1.0/openapi3/schemas/CorrelationId.yaml'
+      required:
+        - userName
+        - password
+        - consentRequestId
+    DFSPAuthorizeRequest:
```
Comment (lewisdaly) [inline src/interface/ttk-thirdparty-api-template.yaml:1306]: For web flow, the user needs to provide an updated list of the accounts they wish to grant consent for: https://github.com/mojaloop/pisp/tree/master/docs/linking#141-web

File: src/interface/ttk-thirdparty-api-template.yaml [hunk 2]
```diff
@@ -970,6 +970,222 @@ paths:
...
+        content:
+          application/json:
+            schema:
+              $ref: '#/components/schemas/DFSPLoginRequest'
+            example:
+              userName: 'dfspa.user.name'
+              password: 'dfspa.user.password'
+              consentRequestId: '3b346cec-47b3-4def-b870-edb255aaf6c3'
+      responses:
+        200:
```
Comment (lewisdaly) [inline src/interface/ttk-thirdparty-api-template.yaml:997]: Perhaps this should contain a list of the accounts available to be linked (all accounts that this User has with the DFSP), and some flag on each account indicating whether or not it is pre-selected based on the `POST /consents#scopes` request body.

That way, we can show the user which accounts they are linking, and they can choose to select one or more accounts.

**Review by lewisdaly — CHANGES_REQUESTED** [review]
Looks good, I think we just need to consider the case where a user agrees on the accounts to be granted access to during their session with the DFSP website

File: src/interface/ttk-thirdparty-api-template.yaml (hunk 2 above)
Comment (eoln) [reply src/interface/ttk-thirdparty-api-template.yaml:997]: @lewisdaly, please take a look at my comments given here: https://github.com/mojaloop/pisp/pull/110#pullrequestreview-571113630

File: src/interface/ttk-thirdparty-api-template.yaml [hunk 3]
```diff
@@ -970,6 +970,222 @@ paths:
...
+          $ref: '../../node_modules/@mojaloop/api-snippets/v1.0/openapi3/responses/index.yaml#/404'
+        405:
+          $ref: '../../node_modules/@mojaloop/api-snippets/v1.0/openapi3/responses/index.yaml#/405'
+        406:
+          $ref: '../../node_modules/@mojaloop/api-snippets/v1.0/openapi3/responses/index.yaml#/406'
+        501:
+          $ref: '../../node_modules/@mojaloop/api-snippets/v1.0/openapi3/responses/index.yaml#/501'
+        503:
+          $ref: '../../node_modules/@mojaloop/api-snippets/v1.0/openapi3/responses/index.yaml#/503'
+  /otp:
```
Comment (eoln) [inline src/interface/ttk-thirdparty-api-template.yaml:1056]:
```suggestion
  /OTP:
```
everywhere we use uppercase, so let do it here also

File: src/interface/ttk-thirdparty-api-template.yaml [hunk 4]
```diff
@@ -970,6 +970,222 @@ paths:
...
+          $ref: '../../node_modules/@mojaloop/api-snippets/v1.0/openapi3/responses/index.yaml#/404'
+        405:
+          $ref: '../../node_modules/@mojaloop/api-snippets/v1.0/openapi3/responses/index.yaml#/405'
+        406:
+          $ref: '../../node_modules/@mojaloop/api-snippets/v1.0/openapi3/responses/index.yaml#/406'
+        501:
+          $ref: '../../node_modules/@mojaloop/api-snippets/v1.0/openapi3/responses/index.yaml#/501'
+        503:
+          $ref: '../../node_modules/@mojaloop/api-snippets/v1.0/openapi3/responses/index.yaml#/503'
+  /otp-secret/{ID}:
```
Comment (eoln) [inline src/interface/ttk-thirdparty-api-template.yaml:1146]:
```suggestion
  /OTPorSecret/{ID}:
```
let use camel case as everywhre

**Review by eoln — CHANGES_REQUESTED** [review]

File: src/interface/ttk-thirdparty-api.yaml [hunk 5]
```diff
@@ -2230,6 +2230,110 @@ paths:
           $ref: '#/paths/~1otp/post/responses/501'
         '503':
           $ref: '#/paths/~1otp/post/responses/503'
+  /store/consentRequests:
```
Comment (eoln) [inline src/interface/ttk-thirdparty-api.yaml:2233]:
```suggestion
/store/consentRequest/{ID} 
```
when we will use a generic storage handler from TTK we don't have to implement the logic of how to extract the ID from the payload.

**Review by eoln — CHANGES_REQUESTED** [review]

File: src/interface/ttk-thirdparty-api-template.yaml [hunk 6]
```diff
@@ -970,6 +970,325 @@ paths:
...
+          $ref: '../../node_modules/@mojaloop/api-snippets/v1.0/openapi3/responses/index.yaml#/404'
+        405:
+          $ref: '../../node_modules/@mojaloop/api-snippets/v1.0/openapi3/responses/index.yaml#/405'
+        406:
+          $ref: '../../node_modules/@mojaloop/api-snippets/v1.0/openapi3/responses/index.yaml#/406'
+        501:
+          $ref: '../../node_modules/@mojaloop/api-snippets/v1.0/openapi3/responses/index.yaml#/501'
+        503:
+          $ref: '../../node_modules/@mojaloop/api-snippets/v1.0/openapi3/responses/index.yaml#/503'
+  /store/consentRequests:
```
Comment (eoln) [inline src/interface/ttk-thirdparty-api-template.yaml:1189]:
```suggestion
  /store/consentRequests/{ID}:
```

File: src/interface/ttk-thirdparty-api-template.yaml (hunk 1 above)
Comment (sridharvoruganti) [reply src/interface/ttk-thirdparty-api-template.yaml:1409]: OK

File: src/interface/ttk-thirdparty-api-template.yaml (hunk 3 above)
Comment (sridharvoruganti) [reply src/interface/ttk-thirdparty-api-template.yaml]: Ok

File: src/interface/ttk-thirdparty-api-template.yaml (hunk 4 above)
Comment (sridharvoruganti) [reply src/interface/ttk-thirdparty-api-template.yaml]: Ok

File: src/interface/ttk-thirdparty-api.yaml (hunk 5 above)
Comment (sridharvoruganti) [reply src/interface/ttk-thirdparty-api.yaml]: ok

File: src/interface/ttk-thirdparty-api-template.yaml (hunk 6 above)
Comment (sridharvoruganti) [reply src/interface/ttk-thirdparty-api-template.yaml]: ok

**Review by eoln — APPROVED** [review]
lgtm

File: src/interface/ttk-thirdparty-api-template.yaml [hunk 7]
```diff
@@ -970,6 +970,313 @@ paths:
...
+      responses:
+        200:
+          description: Response containing consentRequest details
+          content:
+            application/json:
+              schema:
+                $ref: '../../node_modules/@mojaloop/api-snippets/thirdparty/openapi3/schemas/Scope.yaml'
+              example:
+                scopes:
+                  - accountId: dfspa.username.1234
```
Comment (lewisdaly) [inline src/interface/ttk-thirdparty-api-template.yaml:1212]: we will want to display some user readable "account nickname" or something here... I'm not sure if that will be in the scope object or not.

Happy to follow up at a later date, we can always add this field later.

**Review by lewisdaly — APPROVED** [review]

**Pull request merged**